examples/tree_1d_dgsem/elixir_advection_diffusion_implicit_sparse_jacobian.jl

```diff
@@ -0,0 +1,150 @@
+# initial condition based on which the sparsity pattern is computed
+ode_jac_type = semidiscretize(semi_jac_type, tspan)
+u0_ode = ode_jac_type.u0
+du_ode = similar(u0_ode)
+
+###############################################################################
+### Compute the Jacobian sparsity pattern ###
+
+# Only the parabolic part of the `SplitODEProblem` is treated implicitly so we only need the parabolic Jacobian, see
+# https://docs.sciml.ai/DiffEqDocs/stable/types/split_ode_types/#SciMLBase.SplitFunction
+# Although we do sparsity detection on the entire RHS (since semi_jac_type depends on both equations and 
+# equations_parabolic), this is equivalent to doing sparsity detection on the diffusion problem alone
+# (see next section for a demonstration of this)
+
+# Wrap the `Trixi.rhs_parabolic!` function to match the signature `f!(du, u)`, see
+# https://adrianhill.de/SparseConnectivityTracer.jl/stable/user/api/#ADTypes.jacobian_sparsity
+rhs_parabolic_wrapped! = (du_ode, u0_ode) -> Trixi.rhs_parabolic!(du_ode, u0_ode, semi_jac_type, tspan[1])
+
+jac_prototype_parabolic = jacobian_sparsity(rhs_parabolic_wrapped!, du_ode, u0_ode, jac_detector)
+
+# For most efficient solving we also want the coloring vector
+
+coloring_prob = ColoringProblem(; structure = :nonsymmetric, partition = :column)
+coloring_alg = GreedyColoringAlgorithm(; decompression = :direct)
+coloring_result = coloring(jac_prototype_parabolic, coloring_prob, coloring_alg)
+coloring_vec_parabolic = column_colors(coloring_result)
+
+###############################################################################
+### Compare sparsity pattern based on hyperbolic-parabolic ###
+### problem with sparsity pattern of parabolic-only problem ###
+
+# We construct a semidiscretization just as we did previously, but this time with advection_velocity=0
+equations_hyperbolic_no_advection = LinearScalarAdvectionEquation1D(0)
+equations_parabolic_no_advection = LaplaceDiffusion1D(diffusivity(), equations_hyperbolic_no_advection)
+
+semi_jac_type_no_advection = SemidiscretizationHyperbolicParabolic(mesh,
+                                             (equations_hyperbolic_no_advection, 
+                                             equations_parabolic_no_advection),
+                                             initial_condition, solver,
+                                             uEltype = jac_eltype)
+ode_jac_type_no_advection = semidiscretize(semi_jac_type_no_advection, tspan)
+
+# Do sparsity detection on our semidiscretization with advection turned off
+rhs_parabolic_wrapped! = (du_ode, u0_ode) -> Trixi.rhs_parabolic!(du_ode, u0_ode, semi_jac_type_no_advection, tspan[1])
+
+jac_prototype_parabolic_no_advection = jacobian_sparsity(rhs_parabolic_wrapped!, du_ode, u0_ode, jac_detector)
+
+coloring_result_no_advection = coloring(jac_prototype_parabolic_no_advection, coloring_prob, coloring_alg)
+coloring_vec_parabolic_no_advection = column_colors(coloring_result)
```

Comment:
So here it suffices to construct only one jac_type semidiscretization (the one with a = 0)